l1-contracts/contracts/state-transition/chain-interfaces/IExecutor.sol

```diff
@@ -134,6 +147,13 @@ interface IExecutor is IBase {
         CommitBatchInfo[] calldata _newBatchesData
     ) external;
 
+    /// @notice same as `commitBatches` but with the chainId so ValidatorTimelock can sort the inputs.
+    function commitBatchesSharedBridge(
```

Comment:
We should take a look at how this function is implemented, probably need to handle the pubdata there.